test/odbc/psqlodbc/xml.cpp

```diff
@@ -0,0 +1,207 @@
+    bbf_test_setup.CloseStmt();
+    bbf_test_setup.ExecQuery(DropObjectStatement("TABLE", BBF_TABLE_NAME));
+    
+    OdbcHandler pg_test_setup(Drivers::GetDriver(ServerType::PSQL));
+    pg_test_setup.ConnectAndExecQuery(DropObjectStatement("VIEW", PG_VIEW_NAME));
+    pg_test_setup.CloseStmt();
+    pg_test_setup.ExecQuery(DropObjectStatement("TABLE", PG_TABLE_NAME));
+  }
+};
+
+TEST_F(PSQL_DataTypes_Xml, Table_Creation) {
+  createTable(ServerType::MSSQL, BBF_TABLE_NAME, TABLE_COLUMNS);
+
+  const vector<int> BBF_LENGTH_EXPECTED = {10, 0};
+  const vector<int> BBF_PRECISION_EXPECTED = {10, 0};
+  const vector<int> BBF_SCALE_EXPECTED = {0, 0};
+  const vector<string> BBF_NAME_EXPECTED = {"int", "xml"};
+
+  testCommonColumnAttributes(ServerType::MSSQL, BBF_TABLE_NAME, 
+    TABLE_COLUMNS.size(), COL1_NAME, 
+    BBF_LENGTH_EXPECTED, BBF_PRECISION_EXPECTED, 
+    BBF_SCALE_EXPECTED, BBF_NAME_EXPECTED);
+
+  const vector<int> PG_LENGTH_EXPECTED = {4, 255};
+  const vector<int> PG_PRECISION_EXPECTED = {0, 0};
+  const vector<int> PG_SCALE_EXPECTED = {0, 0};
+  const vector<string> PG_NAME_EXPECTED = {"int4", "xml"};
+
+  testCommonColumnAttributes(ServerType::PSQL, PG_TABLE_NAME, 
+    TABLE_COLUMNS.size(), COL1_NAME, 
+    PG_LENGTH_EXPECTED, PG_PRECISION_EXPECTED, 
+    PG_SCALE_EXPECTED, PG_NAME_EXPECTED);
+
+  // Clean-up
+  dropObject(ServerType::PSQL, "TABLE", PG_TABLE_NAME);
+  dropObject(ServerType::MSSQL, "TABLE", BBF_TABLE_NAME);
+}
+
+TEST_F(PSQL_DataTypes_Xml, Insertion_Success) {
+  vector<string> inserted_values = {
+    "NULL",
+    SIMPLE_XML,
+    NORMAL_XML,
+    LARGE_XML,
+  };
+  const int NUMBER_OF_ENTRIES = inserted_values.size();
+
+  createTable(ServerType::MSSQL, BBF_TABLE_NAME, TABLE_COLUMNS);
+
+  testInsertionSuccess(ServerType::MSSQL, BBF_TABLE_NAME, COL1_NAME, inserted_values, inserted_values);
+  insertValuesInTable(ServerType::PSQL, PG_TABLE_NAME, inserted_values, false, NUMBER_OF_ENTRIES);
+
+  inserted_values = duplicateElements(inserted_values);
+
+  verifyValuesInObject(ServerType::PSQL, PG_TABLE_NAME, COL1_NAME, inserted_values, inserted_values);
+  verifyValuesInObject(ServerType::MSSQL, BBF_TABLE_NAME, COL1_NAME, inserted_values, inserted_values);
+
+  dropObject(ServerType::PSQL, "TABLE", PG_TABLE_NAME);
+  dropObject(ServerType::MSSQL, "TABLE", BBF_TABLE_NAME);
+}
+
+TEST_F(PSQL_DataTypes_Xml, Update_Success) {
+  vector<string> inserted_values = {
+    NORMAL_XML
+  };
+
+  vector <string> data_update_values = {
+    SIMPLE_XML
+  };
+
+  createTable(ServerType::MSSQL, BBF_TABLE_NAME, TABLE_COLUMNS);
+
+  testInsertionSuccess(ServerType::MSSQL, BBF_TABLE_NAME, COL1_NAME, inserted_values, inserted_values);
+
+  testUpdateSuccess(ServerType::MSSQL, BBF_TABLE_NAME, COL1_NAME, COL2_NAME, data_update_values, data_update_values);
+  verifyValuesInObject(ServerType::MSSQL, BBF_TABLE_NAME, COL1_NAME, data_update_values, data_update_values);
```

Comment:
Sorry, didn't catch this earlier.

`testUpdateSuccess` already does something similar to `verifyValuesInObject`

`verifyValuesInObject` should be used on the other connection
ie After updating with PG connection, `verifyValuesInObject` on BBF

Reply:
Ah, that makes sense, so I'll switch the two calls to varifyValuesInObject to refer to the other connection type.